### PR TITLE
Change action logs dropdown filter to popup / add check log row button to transaction and charging stations

### PR DESCRIPTION
**Pull request by mahdibenromdhane**

### CODE REVIEW

**Comment by LucasBrazi06** [comment]
Not linked to an issue!

File: src/app/types/Log.ts [hunk 1]
```diff
@@ -17,7 +18,12 @@ export interface Log extends Data {
   detailedMessages: string[];
 }
 
+export interface LogActions extends Data {
+action: ServerAction;
+}
+
 export enum LogButtonAction {
   EXPORT_LOGS = 'export_logs',
+  CHECk_LOGS = 'check_logs',
```
Comment (OliveGerste) [inline src/app/types/Log.ts:27]: k in CHECK should be upper case

File: src/app/shared/dialogs/logs/log-actions-dialog-table-data-source.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+@Injectable()
+export class LogActionsDialogTableDataSource extends DialogTableDataSource<LogActions> {
+  constructor(
+    public spinnerService: SpinnerService,
+    public translateService: TranslateService,
+    private messageService: MessageService,
+    private router: Router,
+    private centralServerService: CentralServerService) {
+    super(spinnerService, translateService);
```
Comment (OliveGerste) [inline src/app/shared/dialogs/logs/log-actions-dialog-table-data-source.ts:23]: messageService, router and centralServerService are not used.

File: src/app/pages/logs/filters/log-action-filter.ts [hunk 3]
```diff
@@ -1,22 +1,31 @@
...
     // Define filter
     const filterDef: TableFilterDef = {
       id: 'action',
       httpId: 'Action',
-      type: FilterType.DROPDOWN,
-      name: 'logs.actions',
+      type: FilterType.DIALOG_TABLE,
+      defaultValue: '',
+      label: '',
+      name: 'logs.action',
```
Comment (OliveGerste) [inline src/app/pages/logs/filters/log-action-filter.ts:15]: I would use plural here

**Comment by mahdibenromdhane** [comment]
> Hi @mahdibenromdhane,
> 
> as always, if you have any questions regarding the feedback, come back to me :)
> 
> * [x]  Table log action is missing in session in progress table
>   image
> * [ ]  Sorting of Actions in dialog does not work
>   image

In this case we only need the log shortcut button for the error list to debug.

**Comment by OliveGerste** [comment]
> > Hi @mahdibenromdhane,
> > as always, if you have any questions regarding the feedback, come back to me :)
> > 
> > * [ ]   Table log action is missing in session in progress table
> >   image
> > * [x]   Sorting of Actions in dialog does not work
> >   image
> 
> In this case we only need the log shortcut button for the error list to debug.

@mahdibenromdhane 
Ah okay. I thought it should be in every list, because it is in history list and in error list. But this is fine then.

**Comment by mahdibenromdhane** [comment]
Also Linked to [This Issue](https://github.com/LucasBrazi06/ev-dashboard/issues/1483)

File: src/app/pages/logs/list/logs-list-table-data-source.ts [hunk 4]
```diff
@@ -40,11 +40,18 @@ export class LogsListTableDataSource extends TableDataSource<Log> {
...
+    private activatedRoute: ActivatedRoute,
     private centralServerService: CentralServerService,
     private datePipe: AppDatePipe) {
     super(spinnerService, translateService);
     // Init
     this.initDataSource();
+    this.activatedRoute.queryParams.subscribe(params => {
+      this.setSearchValue(params['id'] ? params['id'] : '');
+      if (params['fromChargingProfile']) {
+        this.setStaticFilters([{ Action: 'ChargingProfiles|ChargingProfileDelete|ChargingProfileUpdate' }]);
```
Comment (OliveGerste) [inline src/app/pages/logs/list/logs-list-table-data-source.ts:52]: When a static filter is set, it is not possible to filter for other actions properly anymore

File: src/app/shared/dialogs/logs/log-actions-dialog-table-data-source.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      observer.complete();
+    });
+  }
+
+  public buildTableColumnDefs(): TableColumnDef[] {
+    return [
+      {
+        id: 'action',
+        name: 'logs.actions',
+        class: 'text-left col-30p',
```
Comment (OliveGerste) [inline src/app/shared/dialogs/logs/log-actions-dialog-table-data-source.ts:59]: I would make the Action Column bigger.
The title of the dialog is wrong.
image

File: src/app/shared/dialogs/logs/log-actions-dialog.component.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+})
+export class LogActionsDialogComponent extends DialogTableDataComponent<LogActions> {
+  constructor(
+    protected dialogRef: MatDialogRef<LogActionsDialogComponent>,
+    private logActionsDialogTableDataSource: LogActionsDialogTableDataSource,
+    @Inject(MAT_DIALOG_DATA) data: any) {
+    super(data, dialogRef, logActionsDialogTableDataSource);
+    // Default title
+    if (this.title === '') {
+      this.title = 'users.select_users';
```
Comment (OliveGerste) [inline src/app/shared/dialogs/logs/log-actions-dialog.component.ts:20]: Title should be changed.

**Comment by mahdibenromdhane** [comment]
> Seems that the merge with master-qa had issues with cars.
> Could you check and make it work?
> Thanks!

Done :)

**Comment by LucasBrazi06** [comment]
> Done :)

Thanks!

File: src/app/pages/charging-stations/charging-station-limitation/charging-plans/charging-plans.component.ts [hunk 7]
```diff
@@ -97,6 +97,18 @@ export class ChargingPlansComponent implements OnInit, AfterViewInit, OnChanges
...
   }
 
+  public redirectToLog() {
+    this.dialogService.createAndShowYesNoDialog(
+      this.translateService.instant('logs.dialog.redirect.title'),
+      this.translateService.instant('logs.dialog.redirect.confirm'),
+    ).subscribe((response) => {
+      if (response === ButtonType.YES) {
+        this.dialogService.closeAll();
+        this.router.navigate(['/logs'], { queryParams: { id: this.chargingStation.id , fromChargingProfile: true} });
```
[the remark this answers is not in the document]
Comment (LucasBrazi06) [reply src/app/pages/charging-stations/charging-station-limitation/charging-plans/charging-plans.component.ts:107]: Code in the button class for the nth times ;-), you know that already.
And use the TableOpenURLAction for this one, it should do the trick.

File: src/app/pages/charging-stations/in-error/charging-stations-in-error-table-data-source.ts [hunk 8]
```diff
@@ -210,6 +213,16 @@ export class ChargingStationsInErrorTableDataSource extends TableDataSource<Char
...
+      case LogButtonAction.CHECK_LOGS:
+        this.dialogService.createAndShowYesNoDialog(
+          this.translateService.instant('logs.dialog.redirect.title'),
+          this.translateService.instant('logs.dialog.redirect.confirm'),
+        ).subscribe((response) => {
+          if (response === ButtonType.YES) {
+            this.router.navigate(['/logs'], { queryParams: { id: chargingStation.id } });
+          }
+        });
+        break;
```
Comment (LucasBrazi06) [inline src/app/pages/charging-stations/in-error/charging-stations-in-error-table-data-source.ts:225]: Same comment as the button above.

File: src/app/pages/logs/list/logs-list-table-data-source.ts [hunk 9]
```diff
@@ -40,13 +43,31 @@ export class LogsListTableDataSource extends TableDataSource<Log> {
...
     super(spinnerService, translateService);
     // Init
     this.initDataSource();
+    this.initFilters();
   }
 
+  public initFilters() {
+    if (this.activatedRoute && this.activatedRoute.params) {
+      this.activatedRoute.queryParams.subscribe(params => {
+        if (params['fromChargingProfile']) {
```
Comment (LucasBrazi06) [inline src/app/pages/logs/list/logs-list-table-data-source.ts:58]: Remove fromChargingProfile props in the UI and pass all the actions separated by '|' from the charging profiles pop-up.
You can also use the WindowService in window.service.ts file.

File: src/app/pages/logs/list/logs-list-table-data-source.ts [hunk 10]
```diff
@@ -204,7 +225,7 @@ export class LogsListTableDataSource extends TableDataSource<Log> {
         new StartDateFilter().getFilterDef(),
         new EndDateFilter().getFilterDef(),
         new LogLevelTableFilter().getFilterDef(),
-        new LogActionTableFilter().getFilterDef(),
+        this.logActionTableFilter,
```
Comment (LucasBrazi06) [inline src/app/pages/logs/list/logs-list-table-data-source.ts:228]: You can new here, this method is called once ;-)
Same below

File: src/app/pages/logs/table-actions/table-check-logs-action.ts [hunk 11]
```diff
@@ -0,0 +1,13 @@
+import { TableOpenAction } from 'app/shared/table/actions/table-open-action';
+import { LogButtonAction } from 'app/types/Log';
+import { TableActionDef } from 'app/types/Table';
+
+export class TableCheckLogsAction extends TableOpenAction {
```
Comment (LucasBrazi06) [inline src/app/pages/logs/table-actions/table-check-logs-action.ts:5]: Can be replaced or inherits from TableOpenURLAction ?

File: src/app/pages/transactions/history/transactions-history-table-data-source.ts [hunk 12]
```diff
@@ -277,6 +280,15 @@ export class TransactionsHistoryTableDataSource extends TableDataSource<Transact
           actionDef.action(transaction, this.dialog, this.refreshData.bind(this));
         }
         break;
+      case LogButtonAction.CHECK_LOGS:
+        this.dialogService.createAndShowYesNoDialog(
```
Comment (LucasBrazi06) [inline src/app/pages/transactions/history/transactions-history-table-data-source.ts:284]: To replace as commented above.

File: src/app/pages/transactions/in-error/transactions-in-error-table-data-source.ts [hunk 13]
```diff
@@ -287,6 +293,16 @@ export class TransactionsInErrorTableDataSource extends TableDataSource<Transact
           actionDef.action(transaction, this.dialog, this.refreshData.bind(this));
         }
         break;
+      case LogButtonAction.CHECK_LOGS:
+        this.dialogService.createAndShowYesNoDialog(
```
Comment (LucasBrazi06) [inline src/app/pages/transactions/in-error/transactions-in-error-table-data-source.ts:297]: To replace as commented above.

File: src/app/shared/dialogs/logs/log-actions-dialog-table-data-source.ts [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+    public translateService: TranslateService) {
+    super(spinnerService, translateService);
+    // Init
+    this.initDataSource();
+  }
+
+  public loadDataImpl(): Observable<DataResult<LogAction>> {
+    return new Observable((observer) => {
+      const actions: LogAction[] = [];
+      const searchValue = this.getSearchValue().length > 0 ? this.getSearchValue() : '';
```
Comment (LucasBrazi06) [inline src/app/shared/dialogs/logs/log-actions-dialog-table-data-source.ts:27]: Your statement is equivalent to:
const searchValue = this.getSearchValue();
;-)

File: src/app/types/GlobalType.ts [hunk 15]
```diff
@@ -33,7 +34,7 @@ export interface SubjectInfo {
 export interface KeyValue {
   key: string;
   value: string;
-  objectRef?: User|SiteArea|Site|Company|Car|CarCatalog|Asset|RefundReport|ChargingStation|CarMaker;
+  objectRef?: User|SiteArea|Site|Company|Car|CarCatalog|Asset|RefundReport|ChargingStation|CarMaker|CarConverter|LogAction;
```
Comment (LucasBrazi06) [inline src/app/types/GlobalType.ts:37]: Funny to see CarConverter here :-)

File: src/app/types/Log.ts [hunk 16]
```diff
@@ -17,7 +18,12 @@ export interface Log extends Data {
   detailedMessages: string[];
 }
 
+export interface LogAction extends Data {
+  action: ServerAction;
+}
+
 export enum LogButtonAction {
   EXPORT_LOGS = 'export_logs',
+  CHECK_LOGS = 'check_logs',
```
Comment (LucasBrazi06) [inline src/app/types/Log.ts:27]: To be checked if necessary

File: src/assets/i18n/es.json [hunk 17]
```diff
@@ -1021,6 +1021,7 @@
     "search_last_week": "La semana pasada",
     "search_this_month": "Este mes",
     "search_last_month": "El mes pasado",
+    "select_actions": "Seleccionar Acciones",
     "no_logs": "No hay registros que coincidan con los filtros.",
```
Comment (LucasBrazi06) [inline src/assets/i18n/es.json:1025]: I think you missed Spanish and German!
Double check that the 4 files are in sync.
Thanks.

**Pull request merged**